Update PQS docs 2.10.0 (#860)
digital-asset/docs.daml.com: docs/2.10.0/docs/query/pqs-user-guide.rst

   Run 'scribe COMMAND --help[-verbose]' for more information on a command.

Daml Platform Support
^^^^^^^^^^^^^^^^^^^^^

> From version 0.5.0 PQS is built and tested against multiple Daml SDK targets. Each Daml SDK target generates it's own ``.jar`` file. These binaries are present in the published participant-query-store docker images. The container defaults to the lowest targeted Daml SDK. If an operator wants to run PQS against a newer target, the will need to set the containers ``workdir`` at runtime to the desired Daml SDK target.

Reviewer: ```suggestion
From version 0.5.0 PQS is built and tested against multiple Daml SDK targets. Each Daml SDK target generates its own ``.jar`` file. These binaries are present in the published participant-query-store docker images. The container defaults to the lowest targeted Daml SDK. To run PQS against a newer target, you must set the container's ``workdir`` at runtime to the desired Daml SDK target.
```

| ``advance datastore watermark``                                                                                                                    | interactions when the latest consecutive watermark is persisted to the datastore                                               |
+----------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------+

All spans are enriched with contextual information through OpenTelemetry's attributes and events where appropriate. It is advisable to get to know this contextual data. Due to the technical nature of asynchronous and parallel execution, PQS heavily employs `span links <https://opentelemetry.io/docs/specs/otel/overview/#links-between-spans>`__ to highlight causal relationships between independent traces. Modern trace visualisation tools leverage this information to provide a usable representation and navigation through the involved traces.

> Below is an example of causal trace data that spans receipt of a transaction from the Ledger API all the way to it becoming visible by PQS's `Read API <#read-api>`__ in Postgres.

Reviewer: ```suggestion
Below is an example of causal trace data that spans from the receipt of a transaction from the Ledger API to it becoming visible by PQS's `Read API <#read-api>`__ in Postgres.
```

   Parent ID      : 161b5911e7a0ec18
   ID             : 1a66240dfd597654
   Name           : UPDATE scribe.__watermark
   Kind           : Client
   Start time     : 2024-11-06 03:16:43.022737084 +0000 UTC
   End time       : 2024-11-06 03:16:43.023134917 +0000 UTC
   Status code    : Unset
   Status message :
   Attributes:
        -> db.operation: Str(UPDATE)
        -> db.sql.table: Str(__watermark)
        -> db.name: Str(scribe)
        -> db.connection_string: Str(postgresql://postgres-scribe:5432)
        -> server.address: Str(postgres-scribe)
        -> server.port: Int(5432)
        -> db.user: Str(pguser)
        -> db.statement: Str(update __watermark set "offset" = ?, ix = ?;)
        -> db.system: Str(postgresql)

   Span #117
   Trace ID       : 71e67e2420deeef36ef3efacea6399dc
   Parent ID      : 161b5911e7a0ec18
   ID             : f0b24fb074fe41f8
   Name           : commit transaction
   Kind           : Internal
   Start time     : 2024-11-06 03:16:43.023629 +0000 UTC
   End time       : 2024-11-06 03:16:43.024157 +0000 UTC
   Status code    : Unset
   Status message :

.. raw:: html

   </details>

Trace Context Propagation
~~~~~~~~~~~~~~~~~~~~~~~~~

> PQS is an intermediary between a ledger instance and downstream applications that would prefer to access data through SQL rather than in streaming manner from Ledger API directly. Despite forming a pipeline between two data storage systems (Canton and Postgres), PQS stores the `original ledger transaction's trace context <https://docs.daml.com/app-dev/bindings-java/open-tracing.html#continue-spans-across-different-applications>`__ for the purposes of propagation rather than its own. This allows downstream applications to decide for themselves how they want to connect to the original submission's trace (as a child span or as a new trace connected through `span links <https://opentelemetry.io/docs/specs/otel/overview/#links-between-spans>`__).

Reviewer: ```suggestion
PQS is an intermediary between a ledger instance and downstream applications that access data through SQL rather than by streaming directly from the Ledger API. Despite forming a pipeline between two data storage systems (Canton and Postgres), PQS stores the `original ledger transaction's trace context <https://docs.daml.com/app-dev/bindings-java/open-tracing.html#continue-spans-across-different-applications>`__ for propagation. This allows downstream applications to decide for themselves how they want to connect to the original submission's trace (as a child span or as a new trace connected through `span links <https://opentelemetry.io/docs/specs/otel/overview/#links-between-spans>`__).
```
